failed to build air-quality-sensor-app

### Build issue(s)

1. OS Ubuntu WSL2
2. Compiler : gcc (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0

error: format ‘%hu’ expects argument of type ‘int’, but argument 4 has type ‘chip::app::Clusters::AirQuality::AirQualityEnum’ [-Werror=format=]
140 | ChipLogDetail(NotSpecified, "Updated AirQuality value: %huu", newValue);

https://github.com/project-chip/connectedhomeip/blob/d1b59df60a74ac5d73e606619e68752d1c65cb27/examples/air-quality-sensor-app/air-quality-sensor-common/src/air-quality-sensor-manager.cpp#L137

### Platform

other

### Anything else?

root@DESKTOP-HMRIUNJ:/home/bs/test_matter/connectedhomeip/examples/air-quality-sensor-app/linux# gn gen out/debug
Done. Made 272 targets from 223 files in 490ms
root@DESKTOP-HMRIUNJ:/home/bs/test_matter/connectedhomeip/examples/air-quality-sensor-app/linux# ninja -C out/debug
ninja: Entering directory `out/debug'
[468/587] c++ obj/third_party/connectedhomeip/examples/air-quality-senso...ensor-common/src/air-quality-sensor-app.air-quality-sensor-manager.cpp.o
FAILED: obj/third_party/connectedhomeip/examples/air-quality-sensor-app/air-quality-sensor-common/src/air-quality-sensor-app.air-quality-sensor-manager.cpp.o
g++ -MMD -MF obj/third_party/connectedhomeip/examples/air-quality-sensor-app/air-quality-sensor-common/src/air-quality-sensor-app.air-quality-sensor-manager.cpp.o.d -Wconversion -O0 -g2 -fno-common -ffunction-sections -fdata-sections -fno-exceptions -fno-unwind-tables -fno-asynchronous-unwind-tables -fPIC -Wall -Werror -Wextra -Wshadow -Wunreachable-code -Wvla -Wformat -Wformat-nonliteral -Wformat-security -Wundef -Wno-deprecated-declarations -Wno-missing-field-initializers -Wno-unknown-warning-option -Wno-unused-parameter -Wno-cast-function-type -Wno-psabi -Wno-maybe-uninitialized -fdiagnostics-color -fno-strict-aliasing -fmacro-prefix-map=../../third_party/connectedhomeip/= -I/usr/include/gio-unix-2.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -Wno-format-nonliteral -Wno-format-nonliteral -Wno-type-limits -std=gnu++17 -fno-rtti -Wnon-virtual-dtor -DCHIP_HAVE_CONFIG_H=1 -DCHIP_ADDRESS_RESOLVE_IMPL_INCLUDE_HEADER=\<lib/address_resolve/AddressResolve_DefaultImpl.h\> -DCHIP_MINMDNS_USE_EPHEMERAL_UNICAST_PORT=1 -DCHIP_MINMDNS_HIGH_VERBOSITY=0 -DCHIP_MINMDNS_DEFAULT_POLICY=1 -DCHIP_IMGUI_ENABLED=1 -D_REENTRANT -I../../include -I../../third_party/connectedhomeip/examples/air-quality-sensor-app/air-quality-sensor-common/include -I../../third_party/connectedhomeip/zzz_generated/air-quality-sensor-app -Igen/third_party/connectedhomeip/examples/air-quality-sensor-app/air-quality-sensor-common -Igen/third_party/connectedhomeip/examples/air-quality-sensor-app/air-quality-sensor-common/zapgen -I../../third_party/connectedhomeip/src/include -I../../third_party/connectedhomeip/src -Igen/include -I../../third_party/connectedhomeip/zzz_generated/app-common -I../../third_party/connectedhomeip/examples/air-quality-sensor-app/linux/include -I../../third_party/connectedhomeip/config/standalone -I../../third_party/connectedhomeip/third_party/nlassert/repo/include -I../../third_party/connectedhomeip/third_party/nlio/repo/include -I../../third_party/connectedhomeip/third_party/inipp/repo/inipp -I../../third_party/connectedhomeip/src/tracing/perfetto/include -I../../third_party/connectedhomeip/third_party/perfetto/repo/sdk -I../../third_party/connectedhomeip/examples/platform/linux -I../../third_party/connectedhomeip/third_party/jsoncpp/repo/include -I../../third_party/connectedhomeip/examples/common -I../../third_party/connectedhomeip/third_party/imgui/repo -I../../third_party/connectedhomeip/third_party/imgui/repo/backends -I/usr/include/SDL2 -I../../third_party/connectedhomeip/examples/common/QRCode/repo/c -c ../../third_party/connectedhomeip/examples/air-quality-sensor-app/air-quality-sensor-common/src/air-quality-sensor-manager.cpp -o obj/third_party/connectedhomeip/examples/air-quality-sensor-app/air-quality-sensor-common/src/air-quality-sensor-app.air-quality-sensor-manager.cpp.o
In file included from ../../third_party/connectedhomeip/src/lib/support/CodeUtils.h:32,
                 from ../../third_party/connectedhomeip/src/lib/core/CASEAuthTag.h:25,
                 from ../../third_party/connectedhomeip/src/access/SubjectDescriptor.h:23,
                 from ../../third_party/connectedhomeip/src/app/AttributeAccessInterface.h:21,
                 from ../../third_party/connectedhomeip/src/app/clusters/air-quality-server/air-quality-server.h:21,
                 from ../../third_party/connectedhomeip/examples/air-quality-sensor-app/air-quality-sensor-common/include/air-quality-sensor-manager.h:3,
                 from ../../third_party/connectedhomeip/examples/air-quality-sensor-app/air-quality-sensor-common/src/air-quality-sensor-manager.cpp:1:
../../third_party/connectedhomeip/examples/air-quality-sensor-app/air-quality-sensor-common/src/air-quality-sensor-manager.cpp: In member function  void chip::app::Clusters::AirQualitySensorManager::OnAirQualityChangeHandler(chip::app::Clusters::AirQuality::AirQualityEnum)’:
../../third_party/connectedhomeip/examples/air-quality-sensor-app/air-quality-sensor-common/src/air-quality-sensor-manager.cpp:140:33: error: format ‘%hu’ expects argument of type ‘int’, but argument 4 has type ‘chip::app::Clusters::AirQuality::AirQualityEnum’ [-Werror=format=]
  140 |     ChipLogDetail(NotSpecified, "Updated AirQuality value: %huu", newValue);
      |                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~
      |                                                                   |
      |                                                                   chip::app::Clusters::AirQuality::AirQualityEnum
../../third_party/connectedhomeip/src/lib/support/logging/CHIPLogging.h:448:70: note: in definition of macro ‘ChipInternalLogImpl’
  448 |             chip::Logging::Log(chip::Logging::kLogModule_##MOD, CAT, MSG, ##__VA_ARGS__);                                          \
      |                                                                      ^~~
../../third_party/connectedhomeip/src/lib/support/logging/CHIPLogging.h:134:38: note: in expansion of macro ‘ChipInternalLog’
  134 | #define ChipLogDetail(MOD, MSG, ...) ChipInternalLog(MOD, DETAIL, MSG, ##__VA_ARGS__)
      |                                      ^~~~~~~~~~~~~~~
../../third_party/connectedhomeip/examples/air-quality-sensor-app/air-quality-sensor-common/src/air-quality-sensor-manager.cpp:140:5: note: in expansion of macro ‘ChipLogDetail’
  140 |     ChipLogDetail(NotSpecified, "Updated AirQuality value: %huu", newValue);
      |     ^~~~~~~~~~~~~
../../third_party/connectedhomeip/examples/air-quality-sensor-app/air-quality-sensor-common/src/air-quality-sensor-manager.cpp:140:62: note: format string is defined here
  140 |     ChipLogDetail(NotSpecified, "Updated AirQuality value: %huu", newValue);
      |                                                            ~~^
      |                                                              |
      |                                                              int
At global scope:
cc1plus: note: unrecognized command-line option ‘-Wno-unknown-warning-option’ may have been intended to silence earlier diagnostics
cc1plus: all warnings being treated as errors
[473/587] c++ obj/third_party/connectedhomeip/src/protocols/secure_channel/libSecureChannel.CASESession.cpp.o
ninja: build stopped: subcommand failed.

--- a/examples/air-quality-sensor-app/air-quality-sensor-common/src/air-quality-sensor-manager.cpp
+++ b/examples/air-quality-sensor-app/air-quality-sensor-common/src/air-quality-sensor-manager.cpp
@@ -136,7 +136,7 @@ void AirQualitySensorManager::Init()
 
 void AirQualitySensorManager::OnAirQualityChangeHandler(AirQualityEnum newValue)
 {
-    mAirQualityInstance.UpdateAirQuality(static_cast<AirQualityEnum>(newValue));
+    mAirQualityInstance.UpdateAirQuality(newValue);
     ChipLogDetail(NotSpecified, "Updated AirQuality value: %huu", chip::to_underlying(newValue));
 }
 

--- a/examples/air-quality-sensor-app/air-quality-sensor-common/src/air-quality-sensor-manager.cpp
+++ b/examples/air-quality-sensor-app/air-quality-sensor-common/src/air-quality-sensor-manager.cpp
@@ -137,7 +137,7 @@ void AirQualitySensorManager::Init()
 void AirQualitySensorManager::OnAirQualityChangeHandler(AirQualityEnum newValue)
 {
     mAirQualityInstance.UpdateAirQuality(static_cast<AirQualityEnum>(newValue));
-    ChipLogDetail(NotSpecified, "Updated AirQuality value: %huu", newValue);
+    ChipLogDetail(NotSpecified, "Updated AirQuality value: %huu", chip::to_underlying(newValue));
 }
 
 void AirQualitySensorManager::OnCarbonDioxideMeasurementChangeHandler(float newValue)

